airflow/models/xcom_arg.py

```diff
@@ -136,12 +139,15 @@ def set_downstream(
         """Proxy to underlying operator set_downstream method. Required by TaskMixin."""
         self.operator.set_downstream(task_or_task_list, edge_modifier)
 
-    def resolve(self, context: Context) -> Any:
+    @provide_session
+    def resolve(self, context: Context, session: "Session" = NEW_SESSION) -> Any:
```

Comment:
IIRC I wondered if we could do this previously, but eventually did not due to interface compatibility issues.

Reply:
What interface issues might that be?

Reply:
Don’t quite remember now, maybe it has something to do with DagParam also has `resolve`. I guess we can do it and see what happens.